The only purpose of the setuptools_rust runtime dependency is to work around
poetry mishandling build-time dependencies. Eliminate it to avoid spurious
errors when launching synapse.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -174,16 +174,6 @@
 # This is the most recent version of Pydantic with available on common distros.
 pydantic = ">=1.7.4"
 
-# This is for building the rust components during "poetry install", which
-# currently ignores the `build-system.requires` directive (c.f.
-# https://github.com/python-poetry/poetry/issues/6154). Both `pip install` and
-# `poetry build` do the right thing without this explicit dependency.
-#
-# This isn't really a dev-dependency, as `poetry install --no-dev` will fail,
-# but the alternative is to add it to the main list of deps where it isn't
-# needed.
-setuptools_rust = ">=1.3"
-
 
 # Optional Dependencies
 # ---------------------
